src/error.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * Extract a message from a caught error.
+ *
+ * Since you can throw anything in JavaScript,
+ * the given error could be of any type.
+ */
+export function errorMessage(error: unknown): string {
+  const message =
+    typeof error === 'object' && error !== null && 'message' in error
+      ? (error as { message: unknown }).message
+      : error;
+
+  if (message === undefined) {
+    return 'undefined';
+  }
+
+  return typeof message === 'string' ? message : JSON.stringify(error);
+}
```

Comment:
```suggestion
export function errorMessage(error: unknown): string {
  const message = hasMessage(error) ? error.message : error;

  if (message === undefined) {
    return 'undefined';
  }

  return typeof message === 'string' ? message : JSON.stringify(error);
}

function hasMessage(error: unknown): error is { message: unknown } {
  return typeof error === "object" && error !== null && "message" in error;
}
```

Reply:
Better! I wish TypeScript inferred the property access on `message` to be safe after the `in` check.